src/charm.py

```diff
@@ -0,0 +1,161 @@
+            self.on.config_changed,
+            self.on["kubeflow-profiles"].relation_changed,
+        ]:
+            self.framework.observe(event, self.main)
+
+        self.framework.observe(self.on["ingress"].relation_changed, self.configure_mesh)
+
+    def configure_mesh(self, event):
+        if self.interfaces["ingress"]:
+            self.interfaces["ingress"].send_data(
+                {
+                    "prefix": "/",
+                    "rewrite": "/",
+                    "service": self.model.app.name,
+                    "port": self.model.config["port"],
+                }
+            )
+
+    def main(self, event):
+        try:
+            image_details = self.image.fetch()
+        except OCIImageResourceError as e:
+            self.model.unit.status = e.status
+            self.log.info(e)
+            return
+
+        model = self.model.name
+
+        if not (
+            (kf_profiles := self.interfaces["kubeflow-profiles"])
+            and kf_profiles.get_data()
+        ):
+            self.model.unit.status = WaitingStatus(
+                "Waiting for kubeflow-profiles relation data"
+            )
+            return
+
+        kf_profiles = list(kf_profiles.get_data().values())[0]
+        profiles_service = kf_profiles["service-name"]
+
+        port = self.model.config["port"]
+        profile = self.model.config["profile"]
+
+        self.model.unit.status = MaintenanceStatus("Setting pod spec")
+
+        self.model.pod.set_spec(
+            {
+                "version": 2,
+                "serviceAccount": {
+                    "global": True,
+                    "rules": [
+                        {
+                            "apiGroups": [""],
+                            "resources": ["events", "namespaces", "nodes"],
+                            "verbs": ["get", "list", "watch"],
+                        },
+                        {
+                            "apiGroups": ["", "app.k8s.io"],
+                            "resources": [
+                                "applications",
+                                "pods",
+                                "pods/exec",
+                                "pods/log",
+                            ],
+                            "verbs": ["get", "list", "watch"],
+                        },
+                        {"apiGroups": [""], "resources": ["secrets"], "verbs": ["get"]},
+                    ],
+                },
+                "containers": [
+                    {
+                        "name": "kubeflow-dashboard",
+                        "imageDetails": image_details,
+                        "config": {
+                            "USERID_HEADER": "kubeflow-userid",
+                            "USERID_PREFIX": "",
+                            "PROFILES_KFAM_SERVICE_HOST": f"{profiles_service}.{model}",
+                            "REGISTRATION_FLOW": self.model.config["registration-flow"],
```

Comment:
Looks like this section is missing the `DASHBOARD_LINKS_CONFIGMAP`. For right now, having that as an option in `config.yaml` will work, though we might want to consider having that controlled via relations at some point.